bantam-service/src/Bantam/Service/Webship.hs

```diff
@@ -0,0 +1,103 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Bantam.Service.Webship (
+    module X
+  , ResourceDenied (..)
+  , Resource
+  , inconceivable
+  , serverError
+  , lookupAccept
+  , lookupContentType
+  , redirect
+  , forbidden
+  , forbiddenB
+  , halt
+  , resourceToWaiT
+  ) where
+
+import           Bantam.Service.Path
+import           Bantam.Service.View
+
+import           Control.Monad.Trans.Class (lift)
+
+import           Data.ByteString (ByteString)
+import           Data.List (lookup)
+import           Data.List.NonEmpty (NonEmpty (..))
+import qualified Data.List.NonEmpty as NE
+
+import           Network.Wai (Application, Request (..), Response, responseLBS)
+import           Network.Wai as X (requestMethod)
+import           Network.HTTP.Media
+import qualified Network.HTTP.Types as HTTP
+
+import           P
+import qualified Prelude as Unsafe (error)
+
+import           System.IO (IO)
+
+import           Webship.Path (Path)
+import           Webship.Wai (ResponseBody (..), toWaiResponse)
+
+import           X.Control.Monad.Trans.Either (EitherT, hoistEither, runEitherT)
+
+
+data ResourceDenied =
+    UnsupportedAcceptType ByteString
+  | UnsupportedContentType ByteString
+  | Forbidden
+  | NotFound
+
+type Resource m = Request -> EitherT ResourceDenied m Response
```

Comment:
@jystic I hit a wall with webship-resource because you can't set anything in the response. I decided it was easier to bypass that part. It was a huge with that I could keep using path/route without any problems (and checkout how I "secure" a route - I quite like how that fell out).

Despite being a little raw this mini webship-resource version is quite handy.